app/src/main/java/org/wikipedia/views/UserMentionInputView.kt

```diff
@@ -0,0 +1,154 @@
+import org.wikipedia.dataclient.ServiceFactory
+import org.wikipedia.page.Namespace
+import org.wikipedia.util.StringUtil
+import java.util.concurrent.TimeUnit
+
+class UserMentionInputView : LinearLayout, UserMentionEditText.Listener {
+    interface Listener {
+        fun onUserMentionListUpdate()
+        fun onUserMentionComplete()
+    }
+
+    constructor(context: Context) : super(context)
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)
+    constructor(context: Context, attrs: AttributeSet?, defStyle: Int) : super(context, attrs, defStyle)
+
+    val editText get() = binding.inputEditText
+    val textInputLayout get() = binding.inputTextLayout
+    var wikiSite = WikipediaApp.getInstance().wikiSite
+    var listener: Listener? = null
+    var userNameHints: Set<String> = emptySet()
+
+    private val binding = ViewUserMentionInputBinding.inflate(LayoutInflater.from(context), this)
+    private val disposables = CompositeDisposable()
+    private val userNameList = mutableListOf<String>()
+
+    init {
+        orientation = VERTICAL
+        binding.inputEditText.listener = this
+        binding.userListRecycler.layoutManager = LinearLayoutManager(context)
+        binding.userListRecycler.adapter = UserNameAdapter()
+    }
+
+    override fun onDetachedFromWindow() {
+        super.onDetachedFromWindow()
+        disposables.clear()
+    }
+
+    override fun onStartUserNameEntry() {
+        userNameList.clear()
+        binding.userListRecycler.adapter?.notifyDataSetChanged()
+    }
+
+    override fun onCancelUserNameEntry() {
+        disposables.clear()
+        binding.userListRecycler.isVisible = false
+    }
+
+    override fun onUserNameChanged(userName: String) {
+        var userNamePrefix = userName
+        if (userNamePrefix.startsWith("@") && userNamePrefix.length > 1) {
+            userNamePrefix = userNamePrefix.substring(1)
+        }
+        if (userNamePrefix.isNotEmpty()) {
+            searchForUserName(userNamePrefix)
+        }
+    }
+
+    fun maybePrepopulateUserName() {
+        if (binding.inputEditText.text.isNullOrEmpty() && userNameHints.isNotEmpty()) {
+            binding.inputEditText.prepopulateUserName(userNameHints.first())
+        }
+    }
+
+    private fun searchForUserName(prefix: String) {
+        disposables.clear()
+        disposables.add(Observable.timer(200, TimeUnit.MILLISECONDS)
+                .flatMap { ServiceFactory.get(wikiSite).prefixSearchMinimal(prefix, Namespace.USER.code(), 10) }
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({ response ->
+                    userNameList.clear()
+
+                    userNameHints.forEach {
+                        if (it.startsWith(prefix, ignoreCase = true)) {
+                            userNameList.add(it)
+                        }
+                    }
```

Comment:
Would it be possible to simply it as following:
```
userNameList.addAll(userNameHints.filter { it.startsWith(prefix, ignoreCase = true) })
```

Reply:
sure!